setup.py

```diff
@@ -1,18 +1,16 @@
 #!/usr/bin/env python
 
+import errno
+import os
 import subprocess
 import sys
-import os
-import errno
-
-from setuptools import setup
```

Comment:
setuptools should be imported before distutils. it monkey patches distutils.

Reply:
This PR is probably a good opportunity to add a comment to this effect, along with what ever # noqa comment will stop automatic import sorting tools from suggesting this.

On the other hand:   distutils is deprecated and due to be removed in python 3.12, so may need looking in the near future.

Reply:
good point. distutils will be provided by setuptools, so nothing changes there, in theory.

Reply:
I added a comment and turned isort off for this section. 👍